### PR TITLE
capi: Clarify comments

### PR DESCRIPTION
**Pull request by gumb0**

### CODE REVIEW

File: include/fizzy/fizzy.h [hunk 1]
```diff
@@ -165,7 +165,10 @@ bool fizzy_find_exported_function(
     const FizzyModule* module, const char* name, uint32_t* out_func_idx);
 
 /// Instantiate a module.
-/// Takes ownership of module, i.e. @p module is invalidated after this call.
+/// Takes ownership of module, i.e. fizzy_free_module must not be called after this call.
+/// For simplicity module cannot be shared among several instances (calling fizzy_instatiate more
+/// than once with the same module is not allowed), but after fizzy_instantiate functions querying
```
Comment (axic) [inline include/fizzy/fizzy.h:170]: "is not allowed" -> "results in undefined behaviour" ?

File: include/fizzy/fizzy.h [hunk 2]
```diff
@@ -194,7 +197,10 @@ FizzyInstance* fizzy_instantiate(const FizzyModule* module,
     size_t imported_globals_size);
 
 /// Instantiate a module resolving imported functions.
-/// Takes ownership of module, i.e. @p module is invalidated after this call.
+/// Takes ownership of module, i.e. fizzy_free_module must not be called after this call.
+/// For simplicity module cannot be shared among several instances (calling fizzy_instatiate more
+/// than once with the same module is not allowed), but after fizzy_instantiate functions querying
```
Comment (axic) [inline include/fizzy/fizzy.h:202]: Same here?

File: include/fizzy/fizzy.h [hunk 3]
```diff
@@ -165,7 +165,10 @@ bool fizzy_find_exported_function(
     const FizzyModule* module, const char* name, uint32_t* out_func_idx);
 
 /// Instantiate a module.
-/// Takes ownership of module, i.e. @p module is invalidated after this call.
+/// Takes ownership of module, i.e. fizzy_free_module must not be called after this call.
```
Comment (chfast) [inline include/fizzy/fizzy.h:168]:
```suggestion
/// The instance takes ownership of the module, i.e. fizzy_free_module must not be called on the module after this call.
```

File: include/fizzy/fizzy.h [hunk 4]
```diff
@@ -165,7 +165,10 @@ bool fizzy_find_exported_function(
...
+/// For simplicity module cannot be shared among several instances (calling fizzy_instatiate more
```
Comment (chfast) [inline include/fizzy/fizzy.h:169]:
```suggestion
/// For simplicity a module cannot be shared among several instances (calling fizzy_instatiate more
```

**Review by chfast — APPROVED** [review]

**Comment by axic** [comment]
Squashed.

**Pull request merged**